src/main/java/study/step3/domain/RealCar.java

```diff
@@ -16,23 +18,26 @@ public RealCar(String name) {
         this.random = new Random();
     }
 
-    @Override
-    public boolean isMoved() {
-        return moved;
-    }
-
     @Override
     public void move() {
-        moved = guessMove();
+        drivingRecord.add(guessMove());
     }
 
-
     @Override
     public String getName() {
         return name;
     }
 
-    private boolean guessMove() {
+    @Override
+    public int getTotalTry() {
+        return drivingRecord.size();
+    }
+
+    public List<Boolean> takeDrivingRecordTake(int takes) {
+        return drivingRecord.subList(0, takes);
```

Comment:
Collections.unmodifiablelist()를 활용하여 외부에서 값의 변조를 막을 수 없도록 read-only로 리턴하면 좋을 것 같아요!
```suggestion
        return Collections.unmodifiablelist(drivingRecord.subList(0, takes));
```

Reply:
이부분은 한번 생각했었는데 subList 자체가 원본 리스트에 대한 View라서 수정을 가하려고 하면 예외가 발생하게 되어 있어요. 다만 한가지 걸리는 점은 예외 타입이 `ConcurrentModificationException` 라서 예외를 발견했을 때 subList 결과인걸 모르면 원인을 잘 모를 수 있다는 부분이었어요. (실무에서 종종 발생하는 케이스.. 😅)